packages/mds-compliance-engine/engine/mds-compliance-engine.ts

```diff
@@ -84,13 +84,15 @@ export async function createComplianceSnapshot(
 export async function processPolicy(policy: ModalityPolicy, geographies: Geography[]) {
   if (isPolicyActive(policy)) {
     const provider_ids = getProviderIDs(policy.provider_ids)
-    const ComplianceSnapshotPromises = provider_ids.map(async provider_id => {
-      const { filteredEvents, deviceMap } = await getComplianceInputs(provider_id)
+    const getComplianceInputsPromises = provider_ids.map(provider_id => {
+      return getComplianceInputs(provider_id)
+    })
+    const inputs = await Promise.all(getComplianceInputsPromises)
```

Comment:
We probably want to load all of the devices/events/telem into memory once for every provider at start-time (assuming we're doing this all in one container) prior to calling processPolicy, so we can avoid `# of active policies` round trip loads. 

That being said, will be a bunch of stuff to store in memory, but with the concurrency as it is now, we're likely storing far more than that simultaneously anyways.